### PR TITLE
Wet sponge now dries out in nether

### PR DESCRIPTION
**Pull request by x4caa**
issue: https://github.com/df-mc/dragonfly/issues/726

### CODE REVIEW

File: server/block/sponge.go [hunk 1]
```diff
@@ -54,6 +54,13 @@ func (s Sponge) UseOnBlock(pos cube.Pos, face cube.Face, _ mgl64.Vec3, w *world.
 		return
 	}
 
+	// Check if the Sponge is placed in the Nether and if so, turn it into a normal Sponge instantly.
+	if w.Dimension() == world.Nether && s.Wet {
+		s.Wet = false
+		place(w, pos, s, user, ctx)
+		return placed(ctx)
+	}
```
Comment (DaPigGuy) [inline server/block/sponge.go:62]:
```suggestion
	if w.Dimension() == world.Nether && s.Wet {
		s.Wet = false
	}
```

Comment (DaPigGuy) [reply server/block/sponge.go:62]: Additionally, might want to use [this](https://github.com/df-mc/dragonfly/blob/master/server/world/dimension.go#L27) to better support custom dimensions

**Comment by x4caa** [comment]
any idea why the particles are spawning in one of the corners of the block and not the centre?

File: server/block/sponge.go [hunk 2]
```diff
@@ -15,6 +15,8 @@ type Sponge struct {
 
 	// Wet specifies whether the dry or the wet variant of the block is used.
 	Wet bool
+	// Particles specifies whether the block needs particles, only used when a wet sponge is placed in the nether.
+	Particles bool
```
Comment (DaPigGuy) [inline server/block/sponge.go:19]: Why a field?

File: server/block/sponge.go [hunk 3]
```diff
@@ -54,7 +56,16 @@ func (s Sponge) UseOnBlock(pos cube.Pos, face cube.Face, _ mgl64.Vec3, w *world.
...
 
+	// Check if the Sponge is placed in the Nether and if so, turn it into a normal Sponge instantly.
+	if w.Dimension().WaterEvaporates() && s.Wet {
+		s.Wet = false
+		s.Particles = true
+	}
+
 	place(w, pos, s, user, ctx)
+	if s.Particles && placed(ctx) {
+		w.AddParticle(mgl64.Vec3{pos.Vec3().X() + 0.5, pos.Vec3().Y() + 1, pos.Vec3().Z() + 0.5}, particle.Evaporate{})
```
[the remark this answers is not in the document]
Comment (x4caa) [reply server/block/sponge.go:67]: still spawns them in one of the corners

File: server/block/sponge.go (hunk 2 above)
Comment (x4caa) [reply server/block/sponge.go:19]: will change it to a regular variable in my next commit

File: server/block/sponge.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (JustTalDevelops) [reply server/block/sponge.go:67]: Not saying that's why it's happening, just suggesting a simpler way to write that code.

**Pull request merged**